pandas/core/indexes/base.py

```diff
@@ -2110,7 +2110,7 @@ def argsort(self, *args, **kwargs):
         return result.argsort(*args, **kwargs)
 
     def __add__(self, other):
-        return Index(np.array(self) + other)
+        return Index(np.array(self) + other, name=self.name)
```

Comment:
@jreback Thank you for your advice on my first pandas PR. :) Should I raise exception in case of 
```
s = pd.Series([1, 2, 3])
s.index.name = 'foo'
s.index += 1
assert s.index.name == 'foo'
```